### ISSUE TITLE
Reproduction of the paper results

### ISSUE THREAD

**Issue by cornerfarmer** [dingmyu/CamNet#7]
I am not able to reproduce the results on 7-Scenes listed in the paper. 
In my eyes the code should be correct, but the model is totally overfitting on the training data.
While the training loss goes towards zero, the performance of model on the test data stagnates (e.q. mean median translational deviation is always >0.20m)

Is anybody else experiencing similar problems?

**Comment by zhuixunforever** [comment]
Do you reproduce it now? I have the same problem. @cornerfarmer

**Comment by cornerfarmer** [comment]
No, I have given up on that. Let me know, if you get it to work.

**Comment by zhuixunforever** [comment]
Well, I didn't succeed either. Frustrated. Do you have [torchE](https://github.com/dingmyu/CamNet/blob/6039abc7aed10605711bfc9d5ffceb7e9776fb19/Train/pspnet.py#L6), or do you use nn.BatchNorm2d instead of SyncBatchNorm2d ?

**Comment by cornerfarmer** [comment]
I trained on one GPU, but I don't see why this would be a problem

**Comment by westlife35** [comment]
@cornerfarmer  Do you reproduce it now?      or some other new open source method like this (retrieval+relative pose, such as relocnet,camnet) ?

**Comment by cornerfarmer** [comment]
Hey @westlife35,

I have not tried it anymore, so I have no news here.
Regarding other methods: You can take a look at my approach which also works in this way, but is also scene-agnostic: 
https://github.com/DLR-RM/ExReNet

**Comment by westlife35** [comment]
@cornerfarmer hi, I have a question about this code:
In Train/segdata.py, line 71:  
     def quaternion_t(t1, t2):
           return t2-t1
why directly use "t2-t1" as relative translation? why not use the code like this T1-R1*(R2)^(-1)*T2 ? which consider the rotation of images.  I think this may be the reason of cannot reproduction the results of the paper

**Comment by huhulike** [comment]
@westlife35  Hi，I think the author used t2-t1 because the internal parameters of the two cameras are the same by default, so there is no scaling between the camera coordinate systems.
